File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/postgresql/GeometricTypes.scala
package dev.typr.foundationssc.docs.postgresql
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*


import org.postgresql.geometric.{PGcircle, PGpoint, PGpolygon}

@SuppressWarnings(Array("unused"))
object GeometricTypes:
  //start
  val pointType: PgType[PGpoint] = PgTypes.point
  val circleType: PgType[PGcircle] = PgTypes.circle
  val polygonType: PgType[PGpolygon] = PgTypes.polygon

  // Create geometric objects
  val point: PGpoint = new PGpoint(1.0, 2.0)
  val circle: PGcircle = new PGcircle(point, 5.0)
  //stop

File: documentation-examples-kotlin/src/kotlin/dev/typr/foundationskt/docs/postgresql/GeometricTypes.kt
package dev.typr.foundationskt.docs.postgresql

import dev.typr.foundationskt.*
import dev.typr.foundationskt.data.*
import org.postgresql.geometric.PGcircle
import org.postgresql.geometric.PGpoint
import org.postgresql.geometric.PGpolygon

@Suppress("unused")
class GeometricTypes {
    //start
    val pointType: PgType<PGpoint> = PgTypes.point
    val circleType: PgType<PGcircle> = PgTypes.circle
    val polygonType: PgType<PGpolygon> = PgTypes.polygon

    // Create geometric objects
    val point: PGpoint = PGpoint(1.0, 2.0)
    val circle: PGcircle = PGcircle(point, 5.0)
    //stop
}
